### PR TITLE
Bug 1222909 - Update FTEWizard

### PR DESCRIPTION
**Pull request by Fischer-L**
This patch:
- Add option to hide page through css class
- Add option to launch only once
- Attach transition end event on page switching

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3cef62a1a3a5dba2b793faa9410aa6d5f39a6590

File: shared/js/smart-screen/f_t_e_wizard.js [hunk 1]
```diff
@@ -42,6 +42,8 @@
     this._buttonsClass = options.buttonsClass || 'fte-button';
     this._finishButtonClass = options.finishButtonClass || 'fte-finish';
     this._pageClass = options.pageClass || 'fte-page';
+    this._pageHiddenClass = options.pageHiddenClass || null;
```
Comment (rexboy7) [inline shared/js/smart-screen/f_t_e_wizard.js]: Let's give it a default class like `hidden`.

File: shared/js/smart-screen/f_t_e_wizard.js [hunk 2]
```diff
@@ -133,11 +138,19 @@
   };
 
   FTEWizard.prototype._hide = function fw_hide(pageIdx) {
-    this._pages[pageIdx].style.display = 'none';
+    if (this._pageHiddenClass) {
+      this._pages[pageIdx].classList.add(this._pageHiddenClass);
+    } else {
+      this._pages[pageIdx].style.display = 'none';
```
Comment (rexboy7) [inline shared/js/smart-screen/f_t_e_wizard.js]: Let's discard this else case and just use `_pageHiddenClass`.

File: shared/js/smart-screen/f_t_e_wizard.js [hunk 3]
```diff
@@ -133,11 +138,19 @@
...
+    }
   };
 
   FTEWizard.prototype._show = function fw_hide(pageIdx) {
-    this._pages[pageIdx].style.display = '';
+    if (this._pageHiddenClass) {
+      this._pages[pageIdx].classList.remove(this._pageHiddenClass);
+    } else {
+      this._pages[pageIdx].style.display = '';
```
Comment (rexboy7) [inline shared/js/smart-screen/f_t_e_wizard.js]: same above

File: shared/js/smart-screen/f_t_e_wizard.js [hunk 4]
```diff
@@ -42,6 +42,8 @@
     this._buttonsClass = options.buttonsClass || 'fte-button';
     this._finishButtonClass = options.finishButtonClass || 'fte-finish';
     this._pageClass = options.pageClass || 'fte-page';
+    this._pageHiddenClass = options.pageHiddenClass || null;
+    this._onlyLaunchOnce = options.onlyLaunchOnce || true;
```
Comment (rexboy7) [inline shared/js/smart-screen/f_t_e_wizard.js]: can we change to something `false` by default?

Comment (rexboy7) [inline shared/js/smart-screen/f_t_e_wizard.js]: e.g. `launchEveryTime`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=83282fdeb50ea1569e710ac795f2d5dd6f7bd9cd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1f579bd0828e79a2c96a6fe46996dc03cb9ecd91

**Pull request merged**